Yosemite/Yosemite/Stores/CouponStore.swift

```diff
@@ -103,6 +105,27 @@ private extension CouponStore {
             }
         }
     }
+
+    /// Deletes a coupon from a Site with what is persisted in the storage layer.
+    /// After API request succeeds, the stored coupon should be removed from the local storage.
+    /// - Parameters:
+    ///   - siteID: The site that the deleted coupon belongs to.
+    ///   - couponID: The ID of the coupon to be deleted.
+    ///   - onCompletion: Closure to call after deletion is complete. Called on the main thread.
+    ///
+    func deleteCoupon(siteID: Int64, couponID: Int64, onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        remote.deleteCoupon(for: siteID, couponID: couponID) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .failure(let error):
+                onCompletion(.failure(error))
+            case .success:
+                self.deleteStoredCoupon(siteID: siteID, couponID: couponID) {
```

Comment:
nit: do you think it's worth double-checking if the coupon returned by `CoupondsRemote.deleteCoupon` actually matches the given `siteID` and `couponID`?

Reply:
This is definitely worth checking, I added it in 5b2403ed9599bdfd8e28480986d8e959396ca2d8.